### PR TITLE
Quantization Support

### PR DESCRIPTION
**Pull request by saileshd1402**
Changes:
* Added bitsandbytes quantization of precision 4 and 8 bits support. (number of bits can be passed though '-q' argument in llm/run.sh)

### CODE REVIEW

File: llm/kubeflow_inference_run.py [hunk 1]
```diff
@@ -365,6 +369,8 @@ def execute(params: argparse.Namespace) -> None:
     mount_path = params.mount_path
     model_timeout = params.model_timeout
 
+    quantize_bits = params.quantize_bits
```
Comment (gavrishp) [inline llm/kubeflow_inference_run.py:372]: white space on top

File: llm/kubeflow_inference_run.py [hunk 2]
```diff
@@ -382,6 +388,12 @@ def execute(params: argparse.Namespace) -> None:
         model_info["repo_id"] = model_params["repo_id"]
         model_info["repo_version"] = check_if_valid_version(model_info, mount_path)
 
+    if quantize_bits not in [4, 8, 16]:
```
Comment (gavrishp) [inline llm/kubeflow_inference_run.py:391]: 16 is default right as it's no quantization required with 16.

File: llm/kubeflow_inference_run.py (hunk 1 above)
Comment (saileshd1402) [reply llm/kubeflow_inference_run.py:371]: fixed

**Review by gavrishp — APPROVED** [review]

File: llm/handler.py [hunk 3]
```diff
@@ -112,8 +112,29 @@ def initialize(self, context: ts.context.Context):
         self.tokenizer.padding_side = "left"
         logger.info("Tokenizer loaded successfully")
 
+        quantize_bits = self.get_env_value("NAI_QUANTIZATION")
+        quantize_bits = int(quantize_bits) if quantize_bits else quantize_bits
```
Comment (gavrishp) [inline llm/handler.py:116]: keep it simple here.  you are doing the check, just to type cast

```
if self.get_env_value("NAI_QUANTIZATION"):
   quantize_bits = int(self.get_env_value("NAI_QUANTIZATION"))
```

File: llm/kubeflow_inference_run.py [hunk 4]
```diff
@@ -382,6 +387,15 @@ def execute(params: argparse.Namespace) -> None:
         model_info["repo_id"] = model_params["repo_id"]
         model_info["repo_version"] = check_if_valid_version(model_info, mount_path)
 
+    if quantize_bits and int(quantize_bits) not in [4, 8]:
+        print("## Quantization precision bits should be either 4 or 8")
```
Comment (gavrishp) [inline llm/kubeflow_inference_run.py:391]: There can be question, why it's not taking 16 as well. Add a text, 

print("## Quantization precision bits should be either 4 or 8. Default precision used is 16")

Comment (saileshd1402) [reply llm/kubeflow_inference_run.py]: changed the mentioned message

File: llm/handler.py (hunk 3 above)
Comment (saileshd1402) [reply llm/handler.py]: changed as suggested

**Review by gavrishp — APPROVED** [review]

**Review by gavrishp — APPROVED** [review]

**Pull request merged**